### PR TITLE
Bought products

### PR DESCRIPTION
**Pull request by JellyLu**

### CODE REVIEW

File: src/components/enhancer/WithProductList.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                            })
+                        }
+                    </ScrollView>
+                </Screen>
+            )
+        }
+    }
+
+    const mapStateToProps = (state) => {
+        console.log('options.stateName', options.stateName)
```
Comment (JellyLu) [inline src/components/enhancer/WithProductList.tsx:52]: will delete this in next pr

File: src/apis/utils.ts [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
-    },
-  })
-  .then(response => {
-    if (response.status < 400) {
-      return response.json()
-    }
-    throw response
-  })
+export const fetchJson = (url, option) => {
+    const {headers, ...reset} = option
```
Comment (shane123qaz) [inline src/apis/utils.ts:2]: reset or rest ?

File: src/containers/pages/ProfileScreen.tsx [hunk 3]
```diff
@@ -84,13 +84,13 @@ class ProfileScreen extends React.Component<ProfileProps, ProfileStates> {
                     <ButtonWithColor
                         title="已买宝贝"
                         onPress={() => {
-                            this.props.dispatch(NavigationActions.navigate({routeName: 'bought-products'}))
+                            this.props.dispatch(NavigationActions.navigate({routeName: 'boughtProducts'}))
```
Comment (shane123qaz) [inline src/containers/pages/ProfileScreen.tsx:87]: I have a question for navigate. actually we can pass params  like 
NavigationAction.navigate({
  routerName: xxxx
  params: {...}
});

But I don't know how to use params in the other page...

File: src/apis/utils.ts [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
+    return fetch(url, {
+        ...reset,
```
Comment (shane123qaz) [inline src/apis/utils.ts:4]: Not quite understand about why do this change.

File: src/apis/utils.ts (hunk 2 above)
Comment (JellyLu) [reply src/apis/utils.ts:2]: typo

File: src/apis/utils.ts (hunk 4 above)
Comment (JellyLu) [reply src/apis/utils.ts:4]: passed wrong params to fetch. And Only login and logout need  to contain 'Content-Type': 'charset=utf-8', others are only 'application/json; '

**Pull request merged**